### PR TITLE
[VCR, automation] Adds payment intent tests with invalid cards

### PR DESCRIPTION
**Pull request by filipefurtad0**
#### What? Why?

Tests payment intent confirmation errors, when the associated cards are invalid.
 

Since we're now able to record Stripe API calls when creating payment intents, we can create a list of credit cards (each of them with the respective error, as indicated in [Stripe's documentation](https://stripe.com/docs/testing?testing-method=card-numbers#declined-payments)), to request/record/assert payment intent refusals. 

#### What should we test?

- green build

#### Release notes

Changelog Category (reviewers may add a label for the release notes):

- [ ] User facing changes
- [ ] API changes (V0, V1, DFC or Webhook)
- [x] Technical changes only
- [ ] Feature toggled

The title of the pull request will be included in the release notes.

### CODE REVIEW

File: .rubocop_todo.yml [hunk 1]
```diff
@@ -343,6 +343,7 @@ Metrics/ModuleLength:
     - 'spec/services/permissions/order_spec.rb'
     - 'spec/services/variant_units/option_value_namer_spec.rb'
     - 'spec/support/request/stripe_stubs.rb'
+    - 'spec/lib/stripe/payment_intent_validator_spec.rb'
```
Comment (mkllnk) [inline .rubocop_todo.yml:346]: The problem here is that the spec is within the `Stripe` module. You can remove the module and add `Stripe::` where needed, for example the top `describe` block.

**Review by mkllnk — CHANGES_REQUESTED** [review]
Great work! Awesome.

I think that it's better to record the cassettes in the same commit as the change. And it would be good to resolve the style violation. Otherwise all good. :+1:

File: .rubocop_todo.yml (hunk 1 above)
Comment (rioug) [inline .rubocop_todo.yml:346]: Adding it in the rubocop_todo implies we intend to fix it down the line. I think if we are happy to ignore the rule in this case I would use these in the spec file :  
``` ruby
# rubocop:disable Metrics/ModuleLength
....
# rubocop:enable Metrics/ModuleLength

**Review by rioug — APPROVED** [review]
Nice one ! It's good to see we have a more thorough testing of Stripe.

File: .rubocop_todo.yml (hunk 1 above)
Comment (filipefurtad0) [reply .rubocop_todo.yml]: Ah, nice - I did just that on beb47f03c02640f829a41e4cb94ddd0fd747a201.
Thank you.

Comment (filipefurtad0) [reply .rubocop_todo.yml]: > ignore the rule in this case I would use these in the spec file

That makes sense... this way we would not loose track of this rubocop violation. In any case, it won't be necessary this time, I've following Maikel's suggestion above.

**Review by mkllnk — APPROVED** [review]

**Pull request merged**